docs/articles/documentation/test-api/selecting-page-elements/examples-of-working-with-dom-elements.md

```diff
@@ -0,0 +1,135 @@
+---
+layout: docs
+title: Examples of Working with DOM Elements
+permalink: /documentation/test-api/selecting-page-elements/examples-of-working-with-dom-elements.html
+---
+# Examples of Working with DOM Elements
+
+This document shows how to work with DOM elements in frequent real-world situations.
+
+* [Accessing Page Element Properties](#accessing-page-element-properties)
+* [Getting a Page Element Using Custom Logic](#getting-a-page-element-using-custom-logic)
+* [Checking if an Element is Available](#checking-if-an-element-is-available)
+* [Enumerating Elements Identified by a Selector](#enumerating-elements-identified-by-a-selector)
+
+## Accessing Page Element Properties
+
+To work with page elements, use TestCafe [selectors](selectors/README.md).
+Import the `Selector` function and call it passing a CSS selector inside.
+This function creates a selector object [whose API](dom-node-state.md) exposes the most used members of HTML element API.
+
+Note that all property getters are asynchronous, so add the `await` keyword.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `https://devexpress.github.io/testcafe/example/`;
+
+test('My test', async t => {
+    const element     = Selector('#developer-name');
+    const clientWidth = await element.clientWidth;
+
+    // do something with clientWidth
+});
+```
+
+If you need to access an element property not included in the selector's API, request it explicitly by using the [addCustomDOMProperties](selectors/extending-selectors.md) method.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `https://devexpress.github.io/testcafe/example/`;
+
+test('Check Label HTML', async t => {
+    const label = Selector('label').addCustomDOMProperties({
+        innerHTML: el => el.innerHTML
+    });
+
+    await t.expect(label.innerHTML).contains('input type="checkbox" name="remote"');
+});
+```
+
+Note that the `await` keyword can be omitted when specifying a selector property in an [assertion](../assertions/README.md). This activates the [Smart Assertion Query Mechanism](../assertions/README.md#smart-assertion-query-mechanism).
+
+## Getting a Page Element Using Custom Logic
+
+Sometimes CSS selectors are not powerful enough to identify the required page element.
+In this instance, you can introduce a function that picks the desired element.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `https://devexpress.github.io/testcafe/example/`;
+
+test('My test', async t => {
+    const checkBoxesStartingWithR = Selector(() => {
+        var allCheckboxes = document.querySelectorAll('input[type=checkbox]');
+
+        allCheckboxes = Array.prototype.slice.call(allCheckboxes);
+
+        var targetElements = allCheckboxes.filter(function (checkbox) {
```

Comment:
you can use arrow function here

Reply:
@miherlosev seems to be very worried about older browsers. And since we got rid of `NodeList.forEach` because of this, let's not spoil it adding arrow functions.

Reply:
we can use arrow functions in our tests independently of browsers at all

Reply:
If you will use arrow functions then this example will not work in IE11.

Reply:
No. TestCafe allows to use ESNext syntax in the tests (including ClientFunction and Selectors initializers). So arrow functions are ok here. 
Btw, please check all exapmles in chrome, ie, ff, edge

Reply:
Tried an arrow function. the following works in Chrome but it **doesn't** in IE

```js
import { Selector } from 'testcafe';

fixture `My fixture`
    .page `https://devexpress.github.io/testcafe/example/`;

test('My test', async t => {
    const checkBoxesStartingWithR = Selector(() => {
        var allCheckboxes = document.querySelectorAll('input[type=checkbox]');

        allCheckboxes = Array.prototype.slice.call(allCheckboxes);

        var targetElements = allCheckboxes.filter(checkbox => checkbox.labels[0].textContent.startsWith(' R'));

        return targetElements;
    });

    await t.click(checkBoxesStartingWithR.nth(0));
});
```

Reply:
False alarm, this is because of `HTMLInputElement.labels`